docker_mcp/services/stack/validation.py

```
@@ -479,67 +512,85 @@ async def check_port_conflicts(
+                    return True, [], {"ports_checked": [], "conflicts": {}}
+
+                ssh_cmd = build_ssh_command(host)
+                conflicting_ports = []
+                port_details = {}
+
+                for port in ports:
+                    try:
+                        # Check if port is in use using netstat or ss
+                        check_cmd = ssh_cmd + [
+                            f"(netstat -tuln 2>/dev/null | grep ':{port} ' || ss -tuln 2>/dev/null | grep ':{port} ') && echo 'IN_USE' || echo 'AVAILABLE'"
+                        ]
```

Ruff (0.14.4)

Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)